plutus-core/plutus-ir/src/PlutusIR/Compiler.hs

```diff
@@ -241,7 +249,7 @@ compileReadableToPlc (Program a v t) =
         >=> NonStrict.compileNonStrictBindings False
         >=> through check
         >=> (<$ logVerbose "  !!! thunkRecursions")
-        >=> (withBuiltinsInfo . fmap pure . flip ThunkRec.thunkRecursions)
+        >=> (\t' -> withBuiltinsInfo $ \semvar -> withCcOpts $ \opts -> pure $ ThunkRec.thunkRecursions semvar opts t')
```

Comment:
at this point it's easier to just write an inline `\t -> do ...`, I think?

Reply:
this whole thing sucks a bit. I pushed the monadreader compilationctx down to each pass and then I was super excited how the code looked great. But then half of the test cases broke, so i reverted it